app/controllers/course/enrol_requests_controller.rb

```diff
@@ -0,0 +1,132 @@
+class Course::EnrolRequestsController < Course::ModuleController
+  load_and_authorize_resource through: :course
+
+  def index #:nodoc:
+    @staff_requests = @enrol_requests.staff
+    @student_requests = @enrol_requests.student
+
+    respond_to do |format|
+      format.html
+    end
+  end
+
+  def new #:nodoc:
+    unless current_user
```

Comment:
this branch should not be taken because cancancan should block the request. handle this case via authorisation.